src/main/java/org/opentripplanner/middleware/tripMonitor/jobs/CheckMonitoredTrip.java

```diff
@@ -332,19 +440,39 @@ public boolean shouldSkipMonitoredTripCheck() throws Exception {
         // get the configured timezone that OTP is using to parse dates and times
         ZoneId targetZoneId = DateTimeUtils.getOtpZoneId();
 
+        // Make sure journey state indicates that the trip should still possible on at least one day of the week as of
+        // the last time a matching itinerary was attempted to be found. This field in the journey state is used instead
+        // of the trip's itinerary existence data in order to not skip trips that are monitored only one day of the week
+        // and may have not had a matching itinerary on that single monitored day even though it could be possible the
+        // next week.
+        if (previousJourneyState.noLongerPossible) return true;
+
         // get the most recent journey state itinerary to see when the next monitored trip is supposed to occur
-        Itinerary latestMatchingItinerary = trip.journeyState.matchingItinerary;
-        boolean matchingItineraryActiveOrUpcoming = latestMatchingItinerary != null &&
-            latestMatchingItinerary.endTime.after(new Date(trip.journeyState.lastCheckedEpochMillis));
+        boolean matchingItineraryActiveOrUpcoming = previousMatchingItinerary != null &&
+            previousMatchingItinerary.endTime.after(new Date(previousJourneyState.lastCheckedEpochMillis));
         if (matchingItineraryActiveOrUpcoming) {
-            matchingItinerary = latestMatchingItinerary;
-            targetDate = trip.journeyState.targetDate;
+            // skip checking the trip the rest of the time that it is active if it had its trip status marked as the
+            // next trip not being possible
+            if (previousJourneyState.tripStatus == TripStatus.NEXT_TRIP_NOT_POSSIBLE) return true;
```

Comment:
I find it somewhat unclear how this trip status value gets updated if the next trip is not possible. Could that be mentioned here as part of the above comment?

Reply:
Updated in 1987213b6754ffa98f4748820b628be64f9443f4